### PR TITLE
Update maintenance-page example

### PR DESCRIPTION
**Pull request by goncy**
Update maintenance-page example for middleware GA

### CODE REVIEW

File: edge-functions/maintenance-page/middleware.ts [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
 
+// Set pathname were middleware will be executed
+export const config = {
+  matcher: '/big-promo',
+}
+
 export async function middleware(req: NextRequest) {
-  // Clone the URL
-  const url = req.nextUrl.clone()
+  // Filter unwanted paths
```
Comment (dominiksipowicz) [inline edge-functions/maintenance-page/middleware.ts:9]: I think with the matcher this condition is not needed anymore.

**Review by dominiksipowicz — DISMISSED** [review]
`index.tsx` has a snippet block with sample code with old `_middleware`.

https://github.com/vercel/examples/blob/9abb708fe6a56e30dbc1c46a3294a5d4fede2552/edge-functions/maintenance-page/pages/index.tsx#L29

File: edge-functions/maintenance-page/pages/index.tsx [hunk 2]
```diff
@@ -26,26 +26,29 @@ function Home() {
...
+          You can add a <Code>middleware.js</Code> file in the root of your project. Inside <Code>middleware.js</Code> you can do something like
           this:
         </Text>
-        <Snippet>{`import { NextResponse } from 'next/server'
+        <Snippet>{`import { NextRequest, NextResponse } from 'next/server'
 
-export async function middleware(req) {
-  // Clone the URL
-  const url = req.nextUrl.clone()
+// Set pathname were middleware will be executed
```
Comment (lfades) [inline edge-functions/maintenance-page/pages/index.tsx:34]:
```suggestion
```

File: edge-functions/maintenance-page/middleware.ts [hunk 3]
```diff
@@ -1,17 +1,19 @@
 import { NextRequest, NextResponse } from 'next/server'
 
-export async function middleware(req: NextRequest) {
-  // Clone the URL
-  const url = req.nextUrl.clone()
+// Set pathname were middleware will be executed
```
Comment (lfades) [inline edge-functions/maintenance-page/middleware.ts:3]:
```suggestion
```

**Review by lfades — APPROVED** [review]

**Pull request merged**